### PR TITLE
Adjusted app menu item's spacing and padding

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/33315

image

Resolves 

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

`MenuConfigTest.ChangedValueTest`
Click app menu button and check menu bubble's spacing

### CODE REVIEW

File: chromium_src/chrome/browser/ui/views/frame/app_menu_button.cc [hunk 1]
```diff
@@ -3,8 +3,28 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "chrome/browser/ui/views/frame/app_menu_button.h"
```
Comment (fallaciousreasoning) [inline chromium_src/chrome/browser/ui/views/frame/app_menu_button.cc:6]: nit:
```suggestion
#include "chrome/browser/ui/views/frame/app_menu_button.h"

```

File: chromium_src/chrome/browser/ui/views/frame/app_menu_button.cc [hunk 2]
```diff
@@ -3,8 +3,28 @@
...
+  auto& config = const_cast<views::MenuConfig&>(views::MenuConfig::instance());
+  const auto orig_vertical_margin = config.item_vertical_margin;
+  const auto orig_horizontal_padding = config.item_horizontal_padding;
+  config.item_vertical_margin = 9;
+  config.item_horizontal_padding = 16;
+
+  AppMenuButton_ChromiumImpl::RunMenu(std::move(menu_model), browser,
+                                      run_flags);
+
+  config.item_vertical_margin = orig_vertical_margin;
```
Comment (fallaciousreasoning) [inline chromium_src/chrome/browser/ui/views/frame/app_menu_button.cc:28]: Does this mean submenus won't get the padding?

Comment (fallaciousreasoning) [reply chromium_src/chrome/browser/ui/views/frame/app_menu_button.cc:28]: On Linux, they don't seem to:
image

I think @aguscruiz was thinking of changing it globally, so maybe we can just patch the default `MenuConfig`?

Comment (simonhong) [reply chromium_src/chrome/browser/ui/views/frame/app_menu_button.cc]: Oh, thanks. I missed sub menus.
Yeah, changing it globally would be better. Done. 👍🏼

**Review by fallaciousreasoning — APPROVED** [review]
LGTM!

File: patches/ui-views-controls-menu-menu_config.cc.patch [hunk 3]
```diff
@@ -0,0 +1,12 @@
+diff --git a/ui/views/controls/menu/menu_config.cc b/ui/views/controls/menu/menu_config.cc
+index aab2666f7a546ce80446e6d4e83efd6b2b355581..16f004ae674cdfdf442dcc2a012409b013734d5d 100644
+--- a/ui/views/controls/menu/menu_config.cc
++++ b/ui/views/controls/menu/menu_config.cc
+@@ -71,6 +71,7 @@ void MenuConfig::InitCR2023() {
+ // static
+ const MenuConfig& MenuConfig::instance() {
```
Comment (goodov) [inline patches/ui-views-controls-menu-menu_config.cc.patch:7]: I think it's possible to redefine this as `#define instance() instance_ChromiumImpl()` here and in the header and implement a custom `instance()` method.

File: chromium_src/chrome/browser/ui/views/frame/app_menu_button.cc (hunk 2 above)
Comment (aguscruiz) [reply chromium_src/chrome/browser/ui/views/frame/app_menu_button.cc]: Was about to come here and comment just that, yeah it should be global 🙏

File: patches/ui-views-controls-menu-menu_config.cc.patch (hunk 3 above)
Comment (simonhong) [reply patches/ui-views-controls-menu-menu_config.cc.patch]: Done. Thanks!

File: chromium_src/ui/views/controls/menu/menu_config.cc [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+#undef instance
+
+namespace views {
+
+MenuConfig::MenuConfig(const MenuConfig&) = default;
+
+// static
+const MenuConfig& MenuConfig::instance() {
+  auto& config = const_cast<MenuConfig&>(instance_ChromiumImpl());
```
Comment (goodov) [inline chromium_src/ui/views/controls/menu/menu_config.cc:20]: hm. I missed that we'll need `const_cast` 😅. Can we do it like this?
```
const auto& config = instance_ChromiumImpl();
static const bool one_time_call = [](const MenuConfig& config) {
    auto& mutable_config = const_cast<MenuConfig&>(config);
    mutable_config.item_vertical_margin = 9;
    mutable_config.item_horizontal_padding = 16;
    return true;
  }();

return config;
```

**Review by goodov — APPROVED** [review]
lgtm with one possible improvement suggestion

File: chromium_src/ui/views/controls/menu/menu_config.cc (hunk 4 above)
Comment (simonhong) [reply chromium_src/ui/views/controls/menu/menu_config.cc]: Done! 👍🏼

**Pull request merged**